core/object/object.h

```diff
@@ -120,6 +120,7 @@ enum PropertyUsageFlags {
 	PROPERTY_USAGE_EDITOR_BASIC_SETTING = 1 << 27, //for project or editor settings, show when basic settings are selected.
 	PROPERTY_USAGE_READ_ONLY = 1 << 28, // Mark a property as read-only in the inspector.
 	PROPERTY_USAGE_SECRET = 1 << 29, // Export preset credentials that should be stored separately from the rest of the export config.
+	PROPERTY_USAGE_SCRIPT_NON_PUBLIC = 1 << 30, // Used to exclude non-public C# script properties from GDScript autocomplete.
```

Comment:
I think this is too specific for a usage flag. We should probably add non-exported variables to the property list too, as described in #92183: just with the `PROPERTY_USAGE_SCRIPT_VARIABLE` flag, without `PROPERTY_USAGE_DEFAULT`. That would be consistent with GDScript. Not sure about private variables:

* If we exclude private variables, it automatically solves the problem with their autocompletion in GDScript.
* Otherwise, we could use `PROPERTY_USAGE_SCRIPT_VARIABLE | PROPERTY_USAGE_INTERNAL` or add a new flag `PROPERTY_USAGE_PRIVATE` to distinguish between internal and private properties. However, I don't see the point of having private variables in the property list, since they are not accessible.